libraries/ae-inventory-processor/src/main/java/org/metaeffekt/core/inventory/processor/filescan/FileSystemScanExecutor.java

```diff
@@ -136,6 +142,111 @@ public void execute() {
+                    if (foundArtifact != null) {
+                        if (StringUtils.isBlank(assetId)) {
+                            LOG.warn("Cannot resolve asset id for qualifier " + mapper.getQualifier());
+                        } else {
+                            if (!foundArtifact.get(assetId).equals(MARKER_CONTAINS) && !foundArtifact.get(assetId).equals(MARKER_CROSS)) {
+                                LOG.error("Artifact " + foundArtifact.getId() + " does not contain asset " + assetId);
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        // 3. build zips for all components
+        buildZipsForAllComponents(baseDir, filePatternQualifierMappers);
+    }
+
+    public void buildZipsForAllComponents(File baseDir, List<FilePatternQualifierMapper> filePatternQualifierMappers) {
+        // Create an Ant project
+        Project antProject = new Project();
+        antProject.init();
+
+        for (FilePatternQualifierMapper mapper : filePatternQualifierMappers) {
+            // filter out big tar files
+            mapper.getFileMap().values().forEach(files -> files.removeIf(f -> f.length() > 100000000));
+
+            // Loop over each entry in the file map
+            for (Map.Entry<Boolean, List<File>> entry : mapper.getFileMap().entrySet()) {
+                List<File> files = entry.getValue();
+                if (files.isEmpty()) {
+                    continue;
+                }
+
+                final File targetDir = new File(baseDir, "components");
+                if (!targetDir.exists()) {
+                    targetDir.mkdirs();
+                }
+
+                final File tmpFolder = initializeTmpFolder(targetDir);
+
+                final File zipFile = new File(targetDir, mapper.getQualifier() + ".zip");
+                File relativeBaseDir;
+                if (mapper.getPathInAsset().contains(".|\n")) {
+                    relativeBaseDir = new File(baseDir.getPath(), ".");
+                } else {
+                    relativeBaseDir = new File(baseDir.getPath(), mapper.getPathInAsset());
+                }
+
+                // Create a new Ant Zip task
+                Zip zipTask = new Zip();
+                zipTask.setProject(antProject);
+                zipTask.setDestFile(zipFile);
+
+                // Add each file to the zip
+                for (File file : files) {
+                    ZipFileSet zipFileSet = new ZipFileSet();
+
+                    if (file.isFile()) {
+                        zipFileSet.setFile(file);
+                    } else if (file.isDirectory()) {
+                        zipFileSet.setDir(file);
+                    }
+
+                    try {
```

Comment:
This is separate from the fileset creation; so I would not sandwich it in between the fileset creation, but at the end to the for loop as separate aspect with an appropriate single line comment.